tests/test_socket.py

```diff
@@ -0,0 +1,150 @@
+        print(self.id())
+        PexpectTestCase.PexpectTestCase.setUp(self)
+        self.motd = b"""\
+------------------------------------------------------------------------------
+*               Welcome to THE WEATHER UNDERGROUND telnet service!            *
+------------------------------------------------------------------------------
+*                                                                            *
+*   National Weather Service information provided by Alden Electronics, Inc. *
+*    and updated each minute as reports come in over our data feed.          *
+*                                                                            *
+*   **Note: If you cannot get past this opening screen, you must use a       *
+*   different version of the "telnet" program--some of the ones for IBM      *
+*   compatible PC's have a bug that prevents proper connection.              *
+*                                                                            *
+*           comments: jmasters@wunderground.com                              *
+------------------------------------------------------------------------------
+""".replace(b'\n', b'\n\r') + b"\r\n"
+
+    def test_socket(self):
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.connect(('rainmaker.wunderground.com', 23))
+        session = fdpexpect.fdspawn(sock.fileno(), timeout=10)
+        session.expect('Press Return to continue:')
+        self.assertEqual(session.before, self.motd)
+        session.send('\r\n')
+        session.expect('or enter 3 letter forecast city code--')
+        session.send('\r\n')
+        session.expect('Selection:')
+        session.send('X\r\n')
+        session.expect(pexpect.EOF)
+        self.assertEqual(session.before, b'')
+
+    def test_not_int(self):
+        with self.assertRaises(pexpect.ExceptionPexpect):
+            session = fdpexpect.fdspawn('bogus', timeout=10)
+
+    def test_not_file_descriptor(self):
+        with self.assertRaises(pexpect.ExceptionPexpect):
+            session = fdpexpect.fdspawn(-1, timeout=10)
+
+    def test_timeout(self):
+        with self.assertRaises(pexpect.TIMEOUT):
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock.connect(('rainmaker.wunderground.com', 23))
+            session = fdpexpect.fdspawn(sock, timeout=10)
+            session.expect(b'Bogus response')
+
+    def test_interrupt(self):
+        def socket_fn(self):
+            result = 0
+            try:
+                sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+                sock.connect(('rainmaker.wunderground.com', 23))
+                session = fdpexpect.fdspawn(sock, timeout=10)
+                # Get all data from server
+                session.read_nonblocking(size=4096)
+                # This read should timeout
+                session.read_nonblocking(size=4096)
+            except pexpect.TIMEOUT:
+                result = 1
```

Comment:
Are you saying you want to see ECONNREFUSED? Or did you pick 111 out of a hat?
`#define ECONNREFUSED    111     /* Connection refused */`
If you really want something unique, I recommend going above the 255 limit on exit codes:
http://linux.die.net/man/3/exit